### PR TITLE
Upgrade to Rails 6.1

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Keep up to date on dependencies while we're in a workcycle.
Follows on from #2643 and #2625

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require 'active_support/core_ext/integer/time'
```
Comment (jmartin-sul) [inline config/environments/production.rb:3]: was the inclusion of this `require` in prod and test configs but not the dev config intentional, or an oversight?  i'm not totally sure what extensions this provides, but a bit of cursory searching indicates that making the same inclusion in all envs is advised? https://github.com/rails/rails/commit/c8c549750da40cd540ad1064b0b75e5b4e32d3f1

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -3,4 +3,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += %i[
+  passw secret token _key crypt salt certificate otp ssn
```
Comment (jmartin-sul) [inline config/initializers/filter_parameter_logging.rb:7]: no need to keep `:password` in the list?

**Review by jmartin-sul — COMMENTED** [review]
a couple non-blocking questions, but otherwise LGTM

File: config/environments/production.rb (hunk 1 above)
Comment (jcoyne) [reply config/environments/production.rb:3]: Yes, I think this was a merge problem. I'll add that to development.rb too.

**Review by jmartin-sul — APPROVED** [review]
approved, but refraining from merging to allow some testing (just automated integration tests?  manual testing also?)

**Comment by ndushay** [comment]
Holding up for access rights?   or combine???

**Comment by jcoyne** [comment]
No blockers from me. I'm leaving this for the TLs and FR to figure out what order to deploy/test stuff in.

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**